### PR TITLE
Add basic Pointers state

### PR DESCRIPTION
**Pull request by grzegorz-bielski**
While [tinkering with gestures](https://github.com/grzegorz-bielski/pure-tetrio/blob/gesture-controls/frontend/src/main/scala/pureframes/tetrio/game/scenes/gameplay/viewmodel/TapGestureArea.scala) I discovered that I actually need access to sequence of pointer events from current frame. The easiest way to do it was to modify Indigo by adding barebones Pointers state.

While this doesn't resolve the https://github.com/PurpleKingdomGames/indigo/issues/476 or https://github.com/PurpleKingdomGames/indigo/issues/475 it could be a starting point that makes the experimentation a little bit easier.

I'd be nice to merge it before 0.15.0 😄

### CODE REVIEW

File: indigo/project/Misc.scala [hunk 1]
```diff
@@ -27,7 +27,7 @@ object Misc {
     UsefulTask("", "sandboxRun", "Run the sandbox game (fastOptJS + Electron)"),
     UsefulTask("", "perfRun", "Run the perf game (fastOptJS + Electron)"),
     UsefulTask("", "shaderRun", "Run the shader game (fastOptJS + Electron)"),
-    UsefulTask("", "scalafmtCheckAll", "Launch VSCode"),
+    UsefulTask("", "scalafmtCheckAll", "Check formatting"),
```
Comment (davesmith00000) [inline indigo/project/Misc.scala:30]: Thanks! :pray:

File: indigo/indigo/src/main/scala/indigo/shared/input/Pointers.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    Pointers(Batch.empty, Point.zero)
+
+  def calculateNext(previous: Pointers, events: Batch[PointerEvent]) =
+    Pointers(
+      pointerEvents = events,
+      position = lastPointerPosition(previous.position, events)
+    )
+
+  private def lastPointerPosition(previous: Point, events: Batch[PointerEvent]): Point =
+    events.collect { case mp: PointerEvent.PointerMove => mp.position }.lastOption.fold(previous)(identity)
```
Comment (davesmith00000) [inline indigo/indigo/src/main/scala/indigo/shared/input/Pointers.scala:24]: No need to change it, just a comment: I've seen fold used like this a few times recently and I just don't understand how it's syntactically desirable over `getOrElse`. Maybe you can enlighten me on it's appeal?

**Review by davesmith00000 — APPROVED** [review]
Looks like a great start for Pointer state! Thanks for the work. :smile: 

With regard to Indigo 0.15.0, I think I'm nearly there. I'm testing it hard as we speak. Hopefully soon...

File: indigo/indigo/src/main/scala/indigo/shared/input/Pointers.scala (hunk 2 above)
Comment (grzegorz-bielski) [reply indigo/indigo/src/main/scala/indigo/shared/input/Pointers.scala:24]: It was just a straight forward pattern match at first which I directly translated to fold, but yeah the `getOrElse` could have been even better.  I can change it later.

**Pull request merged**